src/main/java/org/whispersystems/textsecuregcm/storage/DirectoryReconciler.java

```diff
@@ -0,0 +1,344 @@
+
+  private static final Logger logger = LoggerFactory.getLogger(DirectoryReconciler.class);
+
+  private static final MetricRegistry metricRegistry      = SharedMetricRegistries.getOrCreate(Constants.METRICS_NAME);
+  private static final Timer          readChunkTimer      = metricRegistry.timer(name(DirectoryReconciler.class, "readChunk"));
+  private static final Timer          sendChunkTimer      = metricRegistry.timer(name(DirectoryReconciler.class, "sendChunk"));
+  private static final Meter          sendChunkErrorMeter = metricRegistry.meter(name(DirectoryReconciler.class, "sendChunkError"));
+
+  private static final String WORKERS_KEY       = "directory_reconciliation_workers";
+  private static final String ACTIVE_WORKER_KEY = "directory_reconciliation_active_worker";
+  private static final String LAST_NUMBER_KEY   = "directory_reconciliation_last_number";
+  private static final String CACHED_COUNT_KEY  = "directory_reconciliation_cached_count";
+  private static final String ACCELERATE_KEY    = "directory_reconciliation_accelerate";
+
+  private static final long   CACHED_COUNT_TTL_MS       = 21600_000L;
+  private static final long   WORKER_TTL_MS             = 120_000L;
+  private static final long   PERIOD                    = 86400_000L;
+  private static final long   CHUNK_INTERVAL            = 42_000L;
+  private static final long   ACCELERATE_CHUNK_INTERVAL = 500L;
+  private static final double JITTER_MAX                = 0.20;
+
+  private final String              serverApiUrl;
+  private final ReplicatedJedisPool jedisPool;
+  private final AccountsManager     accountsManager;
+  private final Client              client;
+
+  private final String       workerId;
+  private final SecureRandom random;
+
+  private final UnlockOperation unlockOperation;
+
+  private boolean running;
+  private boolean finished;
+
+
+  public DirectoryReconciler(ContactDiscoveryConfiguration cdsConfig,
+                             ReplicatedJedisPool jedisPool,
+                             AccountsManager accountsManager)
+      throws IOException
+  {
+    DirectoryConfiguration directoryConfig = cdsConfig.getDirectoryConfiguration();
+    this.serverApiUrl    = directoryConfig.getServerApiUrl();
+    this.jedisPool       = jedisPool;
+    this.accountsManager = accountsManager;
+
+    this.random = new SecureRandom();
+
+    byte[] workerIdBytes = new byte[16];
+    random.nextBytes(workerIdBytes);
+    this.workerId = Hex.toString(workerIdBytes);
+
+    this.unlockOperation = new UnlockOperation(jedisPool);
+
+    SslConfigurator sslConfig = SslConfigurator.newInstance()
+                                               .securityProtocol("TLSv1.2");
+    try {
+      sslConfig.trustStore(initializeKeyStore(directoryConfig.getServerApiCaCertificate()));
+    } catch (CertificateException ex) {
+      logger.error("error reading serverApiCaCertificate from contactDiscovery config", ex);
```

Comment:
Why do we want to continue here?

Reply:
done in 0614c6425b5327bd0f6be6396515384d7e8adb67